ProteoformSuiteInternal/Proteoform.cs

```diff
@@ -166,7 +166,7 @@ public List<ExperimentalProteoform> identify_connected_experimentals(List<PtmSet
                     int mod_rank = Sweet.lollipop.theoretical_database.unlocalized_lookup.TryGetValue(m, out UnlocalizedModification u) ? u.ptm_rank : Sweet.lollipop.modification_ranks.TryGetValue(m.monoisotopicMass, out int x) ? x : Sweet.lollipop.mod_rank_sum_threshold;
 
                     bool could_be_m_retention = m.modificationType == "AminoAcid" && m.motif.ToString() == "M" && theoretical_base.begin == 2 && this.begin == 2 && !ptm_set.ptm_combination.Any(p => p.modification.Equals(m));
-                    bool motif_matches_n_terminus = this.begin - theoretical_base.begin  >= 0 && this.begin - theoretical_base.begin < theoretical_base.sequence.Length && m.motif.ToString() == theoretical_base.sequence[this.begin - theoretical_base.begin].ToString();
+                    bool motif_matches_n_terminus = !mods_in_set.Any(mod => mod.modificationType == "AminoAcid" && mod.motif.ToString() == "M") && this.begin - theoretical_base.begin  >= 0 && this.begin - theoretical_base.begin < theoretical_base.sequence.Length && m.motif.ToString() == theoretical_base.sequence[this.begin - theoretical_base.begin].ToString();
```

Comment:
Could you put the `Any()` expression near the end? It will probably take longest, and that way it won't execute if the others return false.

Reply:
Sounds good